cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/types/Type.kt

```diff
@@ -221,3 +222,14 @@ abstract class Type : Node {
         }
     }
 }
+
+/** A shortcut to return [ObjectType.recordDeclaration], if this is a [ObjectType]. */
+var Type.recordDeclaration: RecordDeclaration?
```

Comment:
Because it's a shortcut to check if this type is an object type and if yes, to retrieve the record declaration. Not all types have a record declaration, so I would prefer this as an extension rather than a property.